### DISCUSSION TITLE
AWS-LAMBDA: no such file or directory, open './src/graphql/schema.graphql'

### DISCUSSION THREAD

**Discussion by alakdam07** [serverless/serverless#10376] [Q&A]
I am practicing Graphql and AWS. I used simple serverless framework also create simple Graphql schema. I deployed the schema(seems like this `graphql.schema` file does not deploy), resolvers to AWS. It successfully create a DynamoDB table and lambda. I can able  make POST/GET request via Graphql Playground by using serverless-offline. But the issue is the the api [end-point]( https://wypmr8bdx4.execute-api.eu-north-1.amazonaws.com/development/api/v1/graphql) does not work. It's show me `internal server error`. I was investigating the issue. From cloud watch I found the local schema which I created the Lambda function did find the `graphql.schema`. This is the error I am getting `"ENOENT: no such file or directory, open './src/graphql/schema.graphql'".`
**This is the lambda error [Image][1]**

**This is my lambda function**

    import { ApolloServer } from 'apollo-server-lambda';
    import { ApolloServerPluginLandingPageGraphQLPlayground } from 'apollo-server-core';
    import runWarm from '../utils/run-warm';
    import fs from 'fs';

    const schema = fs.readFileSync('./src/graphql/schema.graphql', 'utf8'); // This is local my schema 
    import resolvers from '../resolvers';

    const server = new ApolloServer({
      typeDefs: schema,
      resolvers,
      introspection: true,
      plugins: [ApolloServerPluginLandingPageGraphQLPlayground()],
    });

    export default runWarm(
      server.createHandler({
        expressGetMiddlewareOptions: {
          cors: {
            origin: '*',
            credentials: true,
            allowedHeaders: ['Content-Type', 'Origin', 'Accept'],
            optionsSuccessStatus: 200,
          },
        },
      })
    );

**This is my serverless YAML file**

    service: serverless-aws-graphql

    package:
      individually: true

    provider:
      name: aws
      profile: ${env:profile}
      runtime: nodejs14.x
      stage: ${env:stage}
      region: eu-north-1
      timeout: 30
      apiName: ${self:service.name}-${self:provider.stage}
      environment:
        ITEM_TABLE: ${self:service}-items-${self:provider.stage}
      iamRoleStatements:
        - Effect: Allow
          Action:
            - dynamodb:Query
            - dynamodb:Scan
            - dynamodb:GetItem
            - dynamodb:PutItem
            - dynamodb:UpdateItem
            - dynamodb:DeleteItem
          Resource: 'arn:aws:dynamodb:${opt:region, self:provider.region}:*:table/${self:provider.environment.ITEM_TABLE}'

      apiGateway:
        shouldStartNameWithService: true

    custom:
      webpack:
        webpackConfig: ./webpack.config.js
        includeModules: true
        packager: 'npm' # Packager that will be used to package your external modules
      warmup:
        enabled: true
        events:
          - schedule: rate(5 minutes)
        prewarm: true
        concurrency: 1
      prune:
        automatic: true
        number: 5

    functions:
      graphql:
        handler: src/handlers/graphql.default
        events:
          - http:
              path: ${env:api_prefix}/graphql
              method: any
              cors: true

    resources:
      Resources:
        ItemsTable:
          Type: 'AWS::DynamoDB::Table'
          Properties:
            AttributeDefinitions:
              - AttributeName: PK
                AttributeType: S
              - AttributeName: SK
                AttributeType: S
              - AttributeName: GSI1PK
                AttributeType: S
              - AttributeName: GSI1SK
                AttributeType: S
            KeySchema:
              - AttributeName: PK
                KeyType: HASH
              - AttributeName: SK
                KeyType: RANGE
            ProvisionedThroughput:
              ReadCapacityUnits: 1
              WriteCapacityUnits: 1
            GlobalSecondaryIndexes:
              - IndexName: GSI1
                KeySchema:
                  - AttributeName: GSI1PK
                    KeyType: HASH
                  - AttributeName: GSI1SK
                    KeyType: RANGE
                Projection:
                  ProjectionType: ALL
                ProvisionedThroughput:
                  ReadCapacityUnits: 1
                  WriteCapacityUnits: 1
            TableName: ${self:provider.environment.ITEM_TABLE}

    plugins:
      - serverless-webpack
      - serverless-offline
      - serverless-plugin-warmup
      - serverless-dotenv-plugin
      - serverless-prune-plugin

  [1]: https://i.stack.imgur.com/3Xj9c.png

**Comment by zkSoju** [comment]
Have the same error! Curious if you were able to find a solution to this?

**Reply by alakdam07** [reply]
I forget what was the issue. But I think it was import issue. Some how the import was not working as expected in Aws level

**Reply by zkSoju** [reply]
Thanks for the response. By importing, do you mean some thing with the serverless deployment (yaml)? Or a setting on AWS? I’m reading into your recent repo to get some insight, maybe it has to do with the include line you added in the yaml?

**Reply by alakdam07** [reply]
TBH I really don't remember what was the solution. I think I made the solution from both Yaml and as well the .ts file. Try to print the  Yaml file and see what is the output

**Reply by zkSoju** [reply]
Print the yaml?

**Reply by zkSoju** [reply]
Looks like you put the typeDefs inside the handler instead, seems like a bandaid fix.

**Reply by alakdam07** [reply]
You can print it like this: https://www.serverless.com/framework/docs/providers/aws/cli-reference/print.

**Reply by alakdam07** [reply]
Now I remember the issue. I did solution like this src/handlers/graphql.graphql. Some how my default did not work. I did not investigate properly and did hot fix 🙂

**Reply by zkSoju** [reply]
Thanks! Although, I was actually able to solve it on my end by modifying the webpack configuration which was omitting the graphql files for some reason.

**Reply by alakdam07** [reply]
Glad that you fixed your issue. 😊
